nginx.sites.conf

```diff
@@ -74,6 +74,27 @@ server {
     }
 }
 
+server {
+    listen 80;
+    listen 445 ssl;
+
+    server_name insecure.ansible.http.tests;
+
+    ssl_certificate /root/ca/insecure.ansible.http.tests-cert.pem;
+    ssl_certificate_key /root/ca/private/insecure.ansible.http.tests-key.pem;
+
+    ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+    ssl_ciphers ECDHE-RSA-AES128-SHA256:!ECDHE-ECDSA-AES128-SHA;
```

Comment:
I think we're going to need something a bit more nuanced than this single endpoint with these two cipher suites. 

The default cipher suites for our tested clients vary, so some of them will be able to connect to this endpoint without additional configuration.

Reply:
The tests themselves handle it, because they both explicitly declare the cipher to use. Or are you saying that it doesn't really test the explicit use of a cipher as intended, because the default cipher suite in python will include this cipher?

Reply:
> Or are you saying that it doesn't really test the explicit use of a cipher as intended, because the default cipher suite in python will include this cipher?

Correct. This endpoint should use a cipher suite that the client supports, but not by default. Otherwise a successful connection doesn't tell us whether or not the cipher suite selection had any effect.

Additionally, the test should try to connect with the default settings and verify the connection fails. That way we'll know if the client defaults are suitable for testing against this endpoint.

The reverse is actually needed for the "test bad cipher" case. We need an endpoint that works with the default cipher suite selection on the client (verified by the test), but that fails when a cipher is chosen that the server does not support.

Reply:
Ok, think I've got this taken care of.

Reply:
scratch that, I've expanded my testing, and need to work out details of the weak cipher a bit more.